streamlit_extras/echo_expander/__init__.py

```diff
@@ -0,0 +1,143 @@
+import streamlit as st
+# import contextlib
+# from streamlit.echo import ( _get_initial_indent, _get_indent )
+# from typing import List
+# import textwrap
+# import traceback
+
+# @contextlib.contextmanager
+# def echo_expander(title="Show code"):
+#     from streamlit import code, warning, empty, source_util, expander
+#     frame = traceback.extract_stack()[-3]
+#     filename, start_line = frame.filename, frame.lineno
+#     with source_util.open_python_file(filename) as source_file:
+#         source_lines = source_file.readlines()
+#     initial_indent = _get_initial_indent(source_lines[start_line:])
+#     lines_to_display: List[str] = []
+#     for line in source_lines[start_line:]:
+#         indent = _get_indent(line)
+#         if indent is not None and indent < initial_indent:
+#             break
+#         lines_to_display.append(line)
+#     code_string = textwrap.dedent("".join(lines_to_display))
+#     yield
+#     expander(title).code(code_string, "python")
+#     st.markdown('---')
+
+###
+# Extension from echo() in streamlit/echo.py
+###
+
+import contextlib
+import textwrap
+import traceback
+from typing import List
+
+@contextlib.contextmanager
+def echo_expander(code_location="above", expander=True, label="Show code"):
+    """Use in a `with` block to draw some code on the app, then execute it.
+
+    Parameters
+    ----------
+    code_location : "above" or "below"
+        Whether to show the echoed code before or after the results of the
+        executed code block.
+        Default is "above"
+    expander : Boolean
+        Whether the code block should occur in an expander.
+        If False, then same as `st.echo`
+        Defualt is True
+    label : Text
+        If expander is True, then the label for the expander.
+        Default is "Show code"
+
+    Example
+    -------
+
+    >>> with st.echo():
+    >>>     st.write('This code will be printed')
+
+    >>> with st.echo(code_location="below", expander=True, label="Expand to see the code"):
+    >>>     st.write('This code will be printed in an expander')
+
+    """
+
+    from streamlit import empty, source_util
+    from streamlit.echo import _get_indent, _get_initial_indent
+    import streamlit as st_st
```

Comment:
Why `st_st`, and not just use `st` as imported outside of this function?

Reply:
I think you are right that using the st will work. I had originally done this another way and it needed a second st inside the element.